tests/functional/test_functions/00-grep_fail.t

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Test that grep_fail fails if test is not there:
+# Manual test only; to test change MANUAL_TEST to true.
+# You are expecting the second two tests to fail.
+. "$(dirname "$0")/test_header"
```

Comment:
The idea is that the dev who changes this function runs
```
MANUAL_TEST=true etc/bin/run-functional-tests -v tests/functional/test_functions/00-grep_fail.t
```

If anyone can tell me how to test the test functions more elegantly I'd be obliged, but I've spent long enough poking the internet for knowledge of TAP, which turns out to be tricky to search the web for.

I won't be offended if review says to delete this test.

Reply:
TBH I don't think we really need to test this, but it doesn't hurt.

I think you can test for failure by piping the output of `grep_fail` (because when tests pass/fail they just write a line with `ok` or `not ok` to stdout) then checking the test went the way we expected it to.

Reply:
I'll give that a quick go - I was struggling to get the test to work in an automated way.